Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -400,9 +413,107 @@ NWC24::ErrorCode NetKDRequestDevice::KDCheckMail(u32* mail_flag, u32* interval)
   return NWC24::WC24_OK;
 }
 
+NWC24::ErrorCode NetKDRequestDevice::DetermineDownloadTask(u16* entry_index,
+                                                           std::optional<u8>* subtask_id) const
+{
+  // As the scheduler does not tell us which entry to download, we must determine that.
+  // A correct entry is one that hasn't been downloaded the longest compared to other entries.
+  // We first need current UTC.
+  auto time_device =
+      std::static_pointer_cast<NetKDTimeDevice>(GetIOS()->GetDeviceByName("/dev/net/kd/time"));
+  const u32 current_utc = time_device->GetAdjustedUTC();
```

Comment:
This implicitly truncates from `u64` to `u32`. I'm guessing a real IOS does do this check in 32 bits but it might make sense to do this in 64 bits in Dolphin?

There's a couple more places where this could apply too, since you're passing around seconds but store to the file in minutes, maybe just do all the time calculations in 64 bits.